From f10d2d903c7dc7b7d0341a01ea9bd9faf8427320 Mon Sep 17 00:00:00 2001
From: "Du, Changbin" <changbin.du@intel.com>
Date: Fri, 6 Mar 2015 18:35:57 +0800
Subject: [PATCH] usb/rndis: fix issue that rndis is not working

The rndis init function gether_setup_name_default miss to init rx_work
field. This lead the work cannot be executed. The rx_work is to process
received network packages and submit new usb read request. without it
network package cannot deliver to dut side network subsystem.

Change-Id: I461c7238e83675db4bca2caf1a14eb0d71d408f3
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-23611
Signed-off-by: Du, Changbin <changbin.du@intel.com>
---
 drivers/usb/gadget/u_ether.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/gadget/u_ether.c b/drivers/usb/gadget/u_ether.c
index 28b291a..9587c78 100644
--- a/drivers/usb/gadget/u_ether.c
+++ b/drivers/usb/gadget/u_ether.c
@@ -1005,6 +1005,7 @@ struct net_device *gether_setup_name_default(const char *netname)
 	spin_lock_init(&dev->lock);
 	spin_lock_init(&dev->req_lock);
 	INIT_WORK(&dev->work, eth_work);
+	INIT_WORK(&dev->rx_work, process_rx_w);
 	INIT_LIST_HEAD(&dev->tx_reqs);
 	INIT_LIST_HEAD(&dev->rx_reqs);
 
-- 
1.9.1

